http.go

```diff
@@ -92,7 +103,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 				return nil, nil, err
 			}
 		}
-		resp, err := (&http.Client{}).Do(callSummary.HTTPRequest)
+		resp, err := DefaultHTTPClient.Do(callSummary.HTTPRequest)
```

Comment:
I think the default client should be immutable - i.e. it is the default that is used, if no other explicit client is used. Therefore I think we shouldn't be able to set the default at all.

If we want to specify a different client, I think we should make it an optional (i.e. potentially `nil`) member of the `tcclient.Client` struct. There is nothing to stop the implementer creating a single `*http.Client` and using it in all `tcclient.Client` types that it creates directly or indirectly.

Note, you cannot change the default http client in the `net/http` package as far as I am aware. Let me know if I'm wrong. If it were changeable, change should be controlled with a mutex.